File: atividade-4/020_maioresQue/020_maioresQue.js
// Retorna uma sublista de u cujos números sejam maiores que x

const list = [4, 6, 30, 3, 15, 3, 10, 7];
const number = 10;

const maioresQue = (n, xs) => xs.filter((x) => x > n);

console.log(maioresQue(number, list));

File: atividade-4/020_maioresQue/020_maioresQue.hs
module Main where

-- Retorna uma sublista de u cujos números sejam maiores que x

maioresQue :: Int -> [Int] -> [Int]
maioresQue n [] = []
maioresQue n (x:xs)
            | x > n = x:maioresQue n xs
            | otherwise = maioresQue n xs

main = do
  a <- readLn :: IO Int
  b <- readLn :: IO [Int]
  print $ maioresQue a b